pkgdown/_pkgdown.yml

```diff
@@ -21,8 +21,18 @@ template:
     border-radius: 0.5rem
     btn-border-radius: 0.25rem
   includes:
-    in_header:
+    in_header: >
       <script defer data-domain="pkgdown.r-lib.org,all.tidyverse.org" src="https://plausible.io/js/plausible.js"></script>
+      <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap-icons@latest/font/bootstrap-icons.css">
+
+navbar:
+  bg: info
+  type: light
+  structure:
+    right: [github, dark-mode]
+  components:
+    dark-mode:
+      href: "#dark-mode"
```

Comment:
This definitely felt a little hacky and would be nice if pkgdown provided a better abstraction, although I personally like the idea of opting into dark mode controls by adding `dark-mode` to the navbar. But pkgdown would probably prefer to create the markup statically, rather than in the JavaScript as I did here.

It's also worth noting that dark mode only works with Bootstrap 5.3 and pkgdown wouldn't want to include the dark mode dropdown unless BS 5.3 is in use.

Reply:
> It's also worth noting that dark mode only works with Bootstrap 5.3 and pkgdown wouldn't want to include the dark mode dropdown unless BS 5.3 is in use.

Shouldn't pkgdown simply require `bslib >= 0.5.2` which will include https://github.com/rstudio/bslib/commit/a4e487ca9a65860eed1a07ba1df0e3c1c0d449b8 (assumed 0.5.2 is the next release number)?

Reply:
bslib and pkgdown support Bootstrap versions 3, 4 and 5, so requiring the latest bslib wouldn't be sufficient. bslib only includes one copy of each major version of Bootstrap, so it would be informative to check the bslib package version, but a pkgdown site might be using BS3, 4 or 5.

Reply:
I see. Since there are separate template files for the different BS versions in pkgdown and you're only changing the ones for BS5, there shouldn't be any further issue when generating a BS3 or 4 site, no?

Reply:
There are already BS5-specific templates in pkgdown, so we'd probably just add to / modify those and make the assumption that anyone using BS5 wants the most current BS5 (5.3).

Reply:
You should still require `bslib >= 0.5.2` since pkgdown copies the Bootstrap assets from it, I think. Otherwise users might wonder why dark mode isn't working although they opted for BS5 in pkgdown.